clientconn.go

```diff
@@ -133,10 +146,13 @@ func (cc *ClientConn) resetTransport(closeTransport bool) error {
 		if closeTransport {
 			t.Close()
 		}
-		newTransport, err := transport.NewClientTransport(cc.dopts.protocol, cc.target, cc.dopts.authOptions)
+		newTransport, err := transport.NewClientTransportTimeout(cc.dopts.protocol, cc.target, cc.dopts.authOptions, cc.dopts.timeout)
```

Comment:
I was basically mirroring the Dial/DialTimeout approach taken in the net package. If you feel it just adds to bloat, though, I'm happy to just update NewClientTransportto take a timeout.